source/vibe/core/connectionpool.d

```diff
@@ -125,6 +125,42 @@ final class ConnectionPool(Connection)
 		foreach (c; removed_conns)
 			disconnect_callback(c);
 	}
+
+	/** Removes an existing connection from the pool
+
+		Params:
+			conn = connection to remove from the pool
+	*/
+	void remove(Connection conn) @safe
+	{
+		foreach (idx, c; m_connections)
+			if (c is conn)
+			{
+				m_connections = m_connections[0 .. idx] ~ m_connections[idx + 1 .. $];
```

Comment:
Should probably also remove from `m_lockConn` to allow calling `add` later on. However, that would mean that the lock count must be asserted to be zero here, or `LockedConnection` would re-establish the entry with a wrong value.

Reply:
> lock count must be asserted to be zero here

But it is most likely locked, since someone already has a reference to the connection and is trying to remove it from the pool.

Check the updated code.

Reply:
Okay, that should work!